internal/tofu/transform_targets.go

```diff
@@ -73,10 +81,115 @@ func (t *TargetsTransformer) selectTargetedNodes(g *Graph, addrs []addrs.Targeta
 		}
 	}
 
+	targetedOutputNodes := t.getTargetedOutputNodes(targetedNodes, g)
+	for _, outputNode := range targetedOutputNodes {
+		targetedNodes.Add(outputNode)
+	}
+
+	return targetedNodes
+}
+
+func (t *TargetsTransformer) getTargetableNodeResourceAddr(v dag.Vertex) addrs.Targetable {
+	switch r := v.(type) {
+	case GraphNodeResourceInstance:
+		return r.ResourceInstanceAddr()
+	case GraphNodeConfigResource:
+		return r.ResourceAddr()
+	default:
+		// Only resource and resource instance nodes can be targeted.
+		return nil
+	}
+}
+
+// Returns a set of targeted nodes, after excluding resources. An excluded resource
+// is either addressed directly, address indirectly via its container, or it's
+// dependent on an excluded node. The rest are targeted nodes.
+func (t *TargetsTransformer) removeExcludedNodes(g *Graph, excludes []addrs.Targetable) dag.Set {
+	targetedNodes := make(dag.Set)
```

Comment:
im not a huge fan of the naming here. I don't think the explicit opposite of exclude is `targetted` but instead maybe `includedNodes` ? 

I got confused reading this thinking that we're mixing up target and exclude logic here.

Reply:
Is that helpful?

f11054456890284c2edd29bc81f85c39f785981f